src/components/FlowList.vue

```diff
@@ -81,43 +107,46 @@
 
   const DEFAULT_LIMIT = 40
 
-  const search = ref<string>('')
-  const searchDebounced = useDebouncedRef(search, 800)
+  const headerExpanded = ref(false)
+  const selected = ref<string[]>([])
 
-  const page = ref(1)
-  const offset = computed({
-    get: () => (page.value - 1) * DEFAULT_LIMIT,
-    set: (value: number) => {
-      page.value = Math.ceil(value / DEFAULT_LIMIT) + 1
+  const search = ref<string>()
+  const searchDebounced = useDebouncedRef(search, 800)
+  const nameLike = computed({
+    get() {
+      return searchDebounced.value === '' ? undefined : searchDebounced.value
+    },
+    set(value: string | undefined) {
+      searchDebounced.value = value
     },
   })
+
+  const page = useRouteQueryParam<number>('page', NumberRouteParam, 1)
```

Comment:
Adding a "page" query param instead of including the `offset` in the synced filter query params solved a lot of complexity. In a future PR I'd like to abstract this into a wrapper component and composition that can solve for this more broadly.